src/archive/src/main.rs

```diff
@@ -239,6 +259,84 @@ fn write_entry(anchor: AnchorNumber, timestamp: Timestamp, entry: ByteBuf) {
             ))
         }
     });
+    write_entry_internal(anchor_number, timestamp, entry)
+}
+
+/// Fetches, archives and acknowledges a batch of entries.
+/// *Note:* Must be written in a way that nothing breaks on overlapping executions of [fetch_entries].
+async fn fetch_entries() {
+    let ii_canister = with_config(|config| config.ii_canister);
+    let fetch_result: CallResult<(Vec<BufferedEntry>,)> =
+        call(ii_canister, "fetch_entries", ()).await;
+
+    let entries = match fetch_result {
+        Ok((entries,)) => entries,
+        Err((code, message)) => {
+            // failed to fetch entries --> store failure information and exit early
+            store_call_error(CallErrorInfo {
+                time: time(),
+                canister: ii_canister,
+                method: "fetch_entries".to_string(),
+                argument: ByteBuf::from(candid::encode_one(()).unwrap()),
+                rejection_code: code as i32,
+                message,
+            });
+            return;
+        }
+    };
+
+    if entries.is_empty() {
+        // empty fetch is considered successful
+        with_call_info_mut(|info| {
+            info.last_successful_fetch = Some(FetchInfo {
+                timestamp: time(),
+                number_of_entries: 0,
+            })
+        });
+        // nothing to do --> exit early
+        return;
+    }
+
+    let expected_seq_nr = with_log(|log| log.len() as u64);
+    let entries_count = entries.len();
+    let lowest_seq_nr = entries.first().unwrap().sequence_number;
+    let highest_seq_nr = entries.last().unwrap().sequence_number;
+
+    // If this condition is false, the fetched entries have already been archived (by an other invocation of fetch_entries).
+    // This can happen if the fetch interval is too short or on call failures, e.g. if the last
+    // acknowledge message got rejected.
+    // In such cases just the acknowledge is sent again.
+    if lowest_seq_nr >= expected_seq_nr {
+        entries
+            .into_iter()
+            .for_each(|e| write_entry_internal(e.anchor_number, e.timestamp, e.entry));
+    }
+
+    let fetch_result: CallResult<()> =
+        call(ii_canister, "acknowledge_entries", (highest_seq_nr,)).await;
```

Comment:
What do we gain from acknowledging entries?

As far as I understand, II will always have room for a full buffer. Why not simply use it as a circular buffer? In which case do we benefit from having the archive acknowledge the entries it has read? Also, by keeping all entries around on II side (for, say, a couple of hours) we would open the door to other archiving mechanisms (for instance, off-chain) and wouldn't limit it to only this canister